src/uwtools/config/jinja2.py

```diff
@@ -109,7 +109,9 @@ def undeclared_variables(self) -> Set[str]:
 # Public functions
 
 
-def dereference(val: _ConfigVal, context: dict, local: Optional[dict] = None) -> _ConfigVal:
+def dereference(
+    val: _ConfigVal, context: dict, local: Optional[dict] = None, keys: Optional[List[str]] = None
+) -> _ConfigVal:
```

Comment:
The new `keys` argument allows us to keep track of the path of keys that led to a value, in particular to a value that is then removed. Verbose log output would now include a message like
```
[2024-03-30T19:12:44]    DEBUG [dereference] Removing value at: a > b > c
```
given YAML containing
```
a:
  b:
    c: !remove
```